### PR TITLE
Fix #1576 max width in qasm

### LINKED ISSUE #1576: QASM conversion violates `max_width` in tk_SCRATCH_BIT register
According to the docstring of [`circuit_from_qasm`](https://tket.quantinuum.com/api-docs/qasm.html#pytket.qasm.circuit_from_qasm), the user can set a "maximum allowed width of classical registers" with `maxwidth`. However, circuits that generate auxiliar `tk_SCRATCH_BIT` registers seem to skip this constraint.

Here is an example QASM file [telep_Z_Z.txt](https://github.com/user-attachments/files/16978611/telep_Z_Z.txt). To reproduce, run:
```
from pytket.qasm import circuit_from_qasm
pytket_circuit = circuit_from_qasm("telep_Z_Z.txt", maxwidth=64)
```
The resulting `pytket_circuit` has a `tk_SCRATCH_BIT` register with more than 100 bits in it, violating the `maxwidth=64` set by the user.

This is problematic in a pipeline where the pytket circuit is then converted to PHIR to be run in PECOS. For now, as a work around, I am using `rename_units` to convert each `tk_SCRATCH_BIT[idx]`  to  `tk_SCRATCH_BIT_idx[0]`. However, I would expect a better approach to create scratch registers of up to `maxwidth` size, and create new ones when the previous one is filled.

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Fixes https://github.com/CQCL/tket/issues/1576

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/tests/qasm_test_files/longreg.qasm
Comment (cqc-alec) [inline pytket/tests/qasm_test_files/longreg.qasm:1]: This seems too big for a unit test. Can we reduce it to a minimal example that shows the original problem?

Comment (cqc-melf) [reply pytket/tests/qasm_test_files/longreg.qasm:1]: Done in [dca36c6](https://github.com/CQCL/tket/pull/1653/commits/dca36c6b7ab60c49b749b3833c13fd59b5c39bf9)

File: pytket/docs/changelog.rst [hunk 1]
```diff
@@ -14,6 +14,7 @@ Fixes:
 * Fix `symbol_substitution` not preserving opgroups.
 * Remove hardware inefficient circuit construction in `_tk1_to_rzsx`
 * Support converting conditional `RangePredicate`s to QASM.
+* Fix `maxwidth` parameter of `circuit_from_qasm_str`
```
Comment (cqc-alec) [inline pytket/docs/changelog.rst:17]: Also mention the register-resizing fix here?

Comment (cqc-melf) [reply pytket/docs/changelog.rst:17]: Not sure what you mean, do you want me to add more details on how the fix is done?

Comment (cqc-alec) [reply pytket/docs/changelog.rst:17]: We're also adding the `scratch_reg_resize_pass()` to the QASM conversion, might be worth mentioning this.

Comment (cqc-melf) [reply pytket/docs/changelog.rst:17]: done in [4488ded](https://github.com/CQCL/tket/pull/1653/commits/4488ded1b194ddf148a059e3040f73efe4b877ba)

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**